### PR TITLE
Included a termination period and implemented security check

### LINKED ISSUE #8: Problem: payment stream is missing termination period
- When I create a payment stream
- I want to define a termination period
- So that the stream can't be cancelled immediately

When a Payer wants to work with a candidate, they also define a termination period. This period allows the stream to continue working after it has been cancelled, for X amount of time. 

Example: 
- block 0: stream was created with 10 blocks termination period
- block 5: Payer terminates stream
- block 8: Payee claims payment from block 0 to block 8
- block 15: Payee claim payment from block 8 to block 15
- block 17: Payee calls claim function, but fails with "stream was terminated" error

sub: #2

### PR DESCRIPTION
**Pull request by accompany1205**
Resolve #8

### CODE REVIEW

**Comment by accompany1205** [comment]
@markholdex , @georgeciubotaru could you review this PR?

File: contracts/OpenStream.sol [hunk 1]
```diff
@@ -66,4 +87,10 @@ contract OpenStream is ReentrancyGuard, IOpenStream {
         emit TokensClaimed(claimableAmount);
     }
 
+    function terminate() external {
```
Comment (georgeciubotaru) [inline contracts/OpenStream.sol:90]: @accompany1205 Like everyone can call this function, is that correct?

**Review by georgeciubotaru — CHANGES_REQUESTED** [review]
view my comments

File: contracts/OpenStream.sol (hunk 1 above)
Comment (accompany1205) [reply contracts/OpenStream.sol:90]: @georgeciubotaru Only payees can call this function.
When creating open streams, payees address is stored

Comment (georgeciubotaru) [reply contracts/OpenStream.sol:90]: @accompany1205 I cannot see that, it's external and without checks, that means everyone can call it

Comment (accompany1205) [reply contracts/OpenStream.sol:90]: @georgeciubotaru You can check line 73. I used modifier.
You can also find onlyPayee modifier in line47

Comment (accompany1205) [reply contracts/OpenStream.sol]: I added a modifier, onlyPayer to the terminate function.
Could you check it again?

**Review by georgeciubotaru — APPROVED** [review]

**Comment by accompany1205** [comment]
@georgeciubotaru I'm waiting for you to merge my PR.

**Pull request merged**